src/Generators/Microsoft.Gen.Logging/Parsing/TypeSymbolExtensions.cs

```diff
@@ -56,12 +57,29 @@ internal static bool ImplementsIFormattable(this ITypeSymbol sym, SymbolHolder s
     }
 
     internal static bool ImplementsISpanFormattable(this ITypeSymbol sym, SymbolHolder symbols)
-        => symbols.SpanFormattableSymbol != null && sym.ImplementsInterface(symbols.SpanFormattableSymbol);
+        => symbols.SpanFormattableSymbol != null && (sym.ImplementsInterface(symbols.SpanFormattableSymbol) || SymbolEqualityComparer.Default.Equals(sym, symbols.SpanFormattableSymbol));
 
     internal static bool IsSpecialType(this ITypeSymbol typeSymbol, SymbolHolder symbols)
         => typeSymbol.SpecialType != SpecialType.None ||
         typeSymbol.OriginalDefinition.SpecialType != SpecialType.None ||
 #pragma warning disable RS1024
         symbols.IgnorePropertiesSymbols.Contains(typeSymbol);
 #pragma warning restore RS1024
+
+    internal static bool HasCustomToString(this ITypeSymbol type)
+    {
+        ITypeSymbol? current = type;
+        while (current != null && current.SpecialType != SpecialType.System_Object)
+        {
+            if (current.GetMembers("ToString").Where(m => m.Kind == SymbolKind.Method).Cast<IMethodSymbol>().Any(m => m.Parameters.Length == 0))
```

Comment:
Should be also `public`, right?

Reply:
That's enforced by the fact object.ToString is public.

Reply:
My point was that you can define something like this:
```cs
	class MyClass
	{
		private string ToString() => "Hello!";
	}
```

And in that case your code will return `true` even though `MyClass.ToString()` won't be called.

Reply:
Done.